subsys/net/lib/openthread/platform/platform-zephyr.h

```diff
@@ -17,6 +17,49 @@
 
 #include <openthread/instance.h>
 
+/**
+ * Event types used to control open thread.
+ *
+ */
+enum pending_events {
+	PENDING_EVENT_NONE,  /* General event for processing non-event based
```

Comment:
I'm not convinced about this `EVENT_NONE`, it's always _SOMETHING.

I see that this event is used in two cases - when OpenThread notifies about its activity and when alarm shots. How about introducing two separate events:
* `PENDING_EVENT_TASKLET`
* `PENDING_EVENT_ALARM`

and setting them appropriately? We could also try to use them in the future to optimize a bit, and handle radio/alarm/tasklets only when an appropriate event is notified.